Development/nmos/control_protocol_state.h

```diff
@@ -0,0 +1,99 @@
+#ifndef NMOS_CONTROL_PROTOCOL_STATE_H
+#define NMOS_CONTROL_PROTOCOL_STATE_H
+
+#include <map>
+#include "cpprest/json_utils.h"
+#include "nmos/control_protocol_handlers.h"
+#include "nmos/control_protocol_typedefs.h"
+#include "nmos/mutex.h"
+
+namespace slog { class base_gate; }
+
+namespace nmos
+{
+    namespace experimental
+    {
+        struct control_class // NcClassDescriptor
+        {
+            utility::string_t description;
+            nmos::nc_class_id class_id;
+            nmos::nc_name name;
+            web::json::value fixed_role;
+
+            web::json::value properties = web::json::value::array(); // NcPropertyDescriptor array
+            std::vector<method> methods; // NcMethodDescriptor method handler array
+            web::json::value events = web::json::value::array();  // NcEventDescriptor array
```

Comment:
Similar here, these arrays are for the descriptors not the instances:
```suggestion
            web::json::value property_descriptors = web::json::value::array(); // NcPropertyDescriptor array
            std::vector<method> method_descriptors; // NcMethodDescriptor method handler array
            web::json::value event_descriptors = web::json::value::array();  // NcEventDescriptor array
```

Reply:
done